sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -529,6 +529,15 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
        """.stripMargin)
   }
 
+  def inferDateWithLegacyTimeParserError(): Throwable = {
+    new IllegalArgumentException(
```

Comment:
For user-facing errors, it should inherit `SparkThrowable`. cc @gengliangwang @MaxGekk @cloud-fan fyi

Reply:
Yes, please use SparkIllegalArgumentException